airflow/providers/amazon/aws/hooks/s3.py

```diff
@@ -292,18 +302,24 @@ def list_keys(
             'PageSize': page_size,
             'MaxItems': max_items,
         }
-
         paginator = self.get_conn().get_paginator('list_objects_v2')
         response = paginator.paginate(
-            Bucket=bucket_name, Prefix=prefix, Delimiter=delimiter, PaginationConfig=config
+            Bucket=bucket_name,
+            Prefix=prefix,
+            Delimiter=delimiter,
+            PaginationConfig=config,
+            StartAfter=start_after_key,
+        )
+        # JMESPath to query directly on paginated results
+        filtered_response = response.search(
```

Comment:
why are you filtering even when start after is `None`?

Reply:
When `StartAfter` is `None`, `paginator.paginate()` doesn't use it as filter. It filters on `StartAfter` only when it's not a `None`. So, filtering works in both the conditions when it's `None` and not `None`.